src/oods_statement.rs

```diff
@@ -269,6 +302,303 @@ impl MainProof {
         result
     }
 
+    pub fn memory_page_registration_args(&self) -> (RegularMemoryPage, Vec<ContinuousMemoryPage>) {
+        let (pages, _) = self.get_pages_and_products(
+            self.public_input.public_memory.clone(),
+            self.interaction_z,
+            self.interaction_alpha,
+        );
+
+        let regular_page = RegularMemoryPage {
+            page: pages.get(&0).unwrap().clone(),
```

Comment:
I noticed that in several places, but if you know that your code is not going to crash, no need to `pages.get(i).unwrap()`, just do `pages[i]` :P